assets/js/base/components/cart-checkout/local-pickup-select/index.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { RadioControlOption } from '@woocommerce/base-components/radio-control/types';
+import { CartShippingPackageShippingRate } from '@woocommerce/types';
+/**
+ * Internal dependencies
+ */
+import RadioControl from '../../radio-control';
+
+interface LocalPickupSelectProps {
+	title: string;
+	setSelectedOption: ( value: string ) => void;
+	selectedOption: string;
+	pickupLocations: CartShippingPackageShippingRate[];
+	onSelectRate: ( value: string ) => void;
+	renderPickupLocation: (
+		location: CartShippingPackageShippingRate,
+		pickupLocationsCount: number
+	) => RadioControlOption;
+}
+/**
+ * Local pickup select component, used to render a package title and local pickup options.
+ */
+export const LocalPickupSelect = ( {
+	title,
+	setSelectedOption,
+	selectedOption,
+	pickupLocations,
+	onSelectRate,
+	renderPickupLocation,
+}: LocalPickupSelectProps ) => {
+	// Hacky way to check if there are multiple packages, this way is borrowed from  `assets/js/base/components/cart-checkout/shipping-rates-control-package/index.tsx`
+	// We have no built-in way of checking if other extensions have added packages.
+	const multiplePackages =
+		document.querySelectorAll(
+			'.wc-block-components-local-pickup-select .wc-block-components-radio-control'
+		).length > 1;
+	return (
+		<div className="wc-block-components-local-pickup-select">
+			{ multiplePackages && <div>{ title }</div> }
+			<RadioControl
+				onChange={ ( value ) => {
+					setSelectedOption( value );
+					onSelectRate( value );
+				} }
+				selected={ selectedOption }
+				options={ pickupLocations.map( ( location ) =>
+					renderPickupLocation( location, pickupLocations.length )
```

Comment:
I believe we should check the length of the `shippingRates` instead of `pickupLocations`, the same in this line: https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/block.tsx#L152

Reply:
Hey @tarhi-saad I updated some internal testing notes and added one to test for this. Thanks for spotting it!